lib/canvas-sketch.js

```diff
@@ -29,6 +29,38 @@ function getTimeProp (oldManager, newSettings) {
   return newSettings.animate ? { time: oldManager.props.time } : undefined;
 }
 
+/**
+ * The main function of the canvasSketch library, responsible for initializing a sketch.
+ *
+ * @param {Function} sketch - The sketch function. This function should be asynchronous and 
+ *                            receive an object with the properties 'width' and 'height'.
+ * @param {Object} settings - The settings object for the sketch.
```

Comment:
Putting `settings` in brackets will make it optional. I think you'll want to do this for all of the `settings.*` properties as well.

```suggestion
 * @param {Object} [settings] - The settings object for the sketch.
```